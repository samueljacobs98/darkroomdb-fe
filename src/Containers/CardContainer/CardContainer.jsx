import Card from "../../Components/Card/Card";
import "./CardContainer.scss";
// import filmData from "../../assets/mockdata/filmData";

const CardContainer = ({ filmData }) => {
  const getCards = () => {
    return filmData.map((film) => <Card key={film.filmId} film={film} />);
  };
  return <div className="card-container">{filmData && getCards()}</div>;
  // return <div className="card-container">{cards}</div>;
};

export default CardContainer;
