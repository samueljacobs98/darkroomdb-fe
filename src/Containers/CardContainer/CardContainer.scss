@use "../../styles/variables" as *;

.card-container {
  display: flex;
  flex-wrap: wrap;
  gap: $gap;
  justify-content: center;
  @media screen and (min-width: $bp-tablet) {
    justify-content: flex-start;
  }
}
