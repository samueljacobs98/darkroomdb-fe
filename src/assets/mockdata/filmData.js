const filmData = [
  {
    id: 1,
    name: "Kodak ColourPlus 200",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 2,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 3,
    name: "Ilford HP5+",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "black and white",
  },
  {
    id: 4,
    name: "CineStill 800T",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 5,
    name: "Fomapan 200",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 6,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 7,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 8,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 9,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 10,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 11,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
  {
    id: 12,
    name: "Kodak Portra 400",
    ratings: [4, 4.5, 5, 4.5, 5, 4.5],
    info: "CineStill 800T is a 35mm tungsten-balanced colour negative film made from Kodak motion picture film stock. Its 800 ISO emulsion is perfect for difficult low light situations. It is a highly flexible film that can be pushed up to 3200 ISO. This emulsion is ideal for scanning.",
    price: "£7.99",
    format: ["35mm", "120"],
    style: "colour",
  },
];

export default filmData;
