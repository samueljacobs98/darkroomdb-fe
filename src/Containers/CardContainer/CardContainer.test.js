import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CardContainer from "./CardContainer";
import filmData from "../../assets/mockdata/filmData";

it("should match component snapshot", () => {
  const { container } = render(
    <MemoryRouter>
      <CardContainer filmData={filmData} />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
