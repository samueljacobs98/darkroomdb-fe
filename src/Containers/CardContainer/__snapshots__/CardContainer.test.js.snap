// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match component snapshot 1`] = `
<div>
  <div
    class="card-container"
  >
    <a
      href="/film/undefined"
    >
      <div
        class="card"
      >
        Kodak ColourPlus 200
      </div>
    </a>
    <a
      href="/film/undefined"
    >
      <div
        class="card"
      >
        Kodak Portra 400
      </div>
    </a>
    <a
      href="/film/undefined"
    >
      <div
        class="card"
      >
        Ilford HP5+
      </div>
    </a>
    <a
      href="/film/undefined"
    >
      <div
        class="card"
      >
        CineStill 800T
      </div>
    </a>
    <a
      href="/film/undefined"
    >
      <div
        class="card"
      >
        Fomapan 200
      </div>
    </a>
    <a
      href="/film/undefined"
    >
      <div
        class="card"
      >
        Kodak Portra 400
      </div>
    </a>
    <a
      href="/film/undefined"
    >
      <div
        class="card"
      >
        Kodak Portra 400
      </div>
    </a>
    <a
      href="/film/undefined"
    >
      <div
        class="card"
      >
        Kodak Portra 400
      </div>
    </a>
    <a
      href="/film/undefined"
    >
      <div
        class="card"
      >
        Kodak Portra 400
      </div>
    </a>
    <a
      href="/film/undefined"
    >
      <div
        class="card"
      >
        Kodak Portra 400
      </div>
    </a>
    <a
      href="/film/undefined"
    >
      <div
        class="card"
      >
        Kodak Portra 400
      </div>
    </a>
    <a
      href="/film/undefined"
    >
      <div
        class="card"
      >
        Kodak Portra 400
      </div>
    </a>
  </div>
</div>
`;
